src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -374,1042 +376,665 @@
 import { mapGetters, mapState } from 'vuex'
 
-// Components
-import ColinFiling from './FilingHistoryList/ColinFiling.vue'
+// Components and Dialogs
 import CompletedAlteration from './FilingHistoryList/CompletedAlteration.vue'
 import CompletedIa from './FilingHistoryList/CompletedIa.vue'
+import FiledLabel from './FilingHistoryList/FiledLabel.vue'
 import FutureEffective from './FilingHistoryList/FutureEffective.vue'
 import FutureEffectivePending from './FilingHistoryList/FutureEffectivePending.vue'
 import PaperFiling from './FilingHistoryList/PaperFiling.vue'
 import PendingFiling from './FilingHistoryList/PendingFiling.vue'
+import StaffFiling from './FilingHistoryList/StaffFiling.vue'
 import { DetailsList } from '@/components/common'
-
-// Dialogs
 import { AddCommentDialog, DownloadErrorDialog, LoadCorrectionDialog } from '@/components/dialogs'
 
 // Enums and Interfaces
-import { FilingStatus, FilingTypes, Routes } from '@/enums'
-import { CourtOrderIF, FilingIF, HistoryItemIF } from '@/interfaces'
+import { DocumentTypes, FilingTypes, Routes } from '@/enums'
+import { HistoryItemIF, LedgerIF } from '@/interfaces'
 
 // Mixins
-import { DateMixin, EnumMixin, FilingMixin, LegalApiMixin } from '@/mixins'
-
-export default {
-  name: 'FilingHistoryList',
-
-  mixins: [DateMixin, EnumMixin, FilingMixin, LegalApiMixin],
+import { DateMixin, EnumMixin, FilingMixin, LegalApiMixin, PayApiMixin } from '@/mixins'
 
+@Component({
   components: {
     // sub-components
-    ColinFiling,
     CompletedAlteration,
     CompletedIa,
+    FiledLabel,
     FutureEffective,
     FutureEffectivePending,
     PaperFiling,
     PendingFiling,
+    StaffFiling,
+
     DetailsList,
+
     // dialogs
     AddCommentDialog,
     DownloadErrorDialog,
     LoadCorrectionDialog
   },
-
-  data () {
-    return {
-      addCommentDialog: false,
-      downloadErrorDialog: false,
-      loadCorrectionDialog: false,
-      panel: null as number, // currently expanded panel
-      historyItems: [] as Array<HistoryItemIF>,
-      loadingDocument: false,
-      loadingReceipt: false,
-      loadingAll: false,
-      currentFilingId: null as number,
-      downloadingDocIndex: -1,
-      FilingTypes // enum for template
-    }
-  },
-
-  props: {
-    disableChanges: null
-  },
-
   computed: {
+    // Property definitions for runtime environment.
     ...mapGetters(['isBComp', 'isRoleStaff', 'nrNumber']),
+    ...mapState(['filings', 'entityName'])
+  }
+})
+export default class FilingHistoryList extends Mixins(
+  DateMixin, EnumMixin, FilingMixin, LegalApiMixin, PayApiMixin
+) {
+  @Prop({ default: false })
+  readonly disableChanges: boolean
+
+  // enums for template
+  readonly DocumentTypes = DocumentTypes
+  readonly FilingTypes = FilingTypes
+
+  readonly isBComp!: boolean
+  readonly isRoleStaff!: boolean
+  readonly nrNumber!: string
+  readonly filings!: Array<LedgerIF>
+  readonly entityName!: string
```

Comment:
Are these computed properties just declared on the class? Is this the alternative because we're not using vuex-class decorators here?
Sorry my memory is short 😆

Reply:
This is the alternate syntax for getters (which are declared in the component declaration. Actually I prefer the other syntax so I'll change it :)

Reply:
Updated.